File: exploration/com/leetcode/LengthOfLongestSubstring.py
class Solution(object):
    def lengthOfLongestSubstring(self, s) -> int:
        if len(s) == 1:
            return 1
        mid, target = "", ""
        for i, x in enumerate(s):
            if (mid.find(x) >= 0):
                if (len(mid) > len(target)):
                    target = mid
                mid = mid[mid.index(x) + 1:] + x
            else:
                mid = mid + x
        return max(len(mid), len(target))


s = Solution()
print(s.lengthOfLongestSubstring("dvdf"))

File: src/main/java/com/universe/leetCode/LengthOfLongestSubstring.java
package com.universe.leetCode;

/**
 * @Description 3题
 * 给定一个字符串，请你找出其中不含有重复字符的 最长子串 的长度。
 * @Author naikuo
 * @Date 2020/2/11 5:39 PM
 */
public class LengthOfLongestSubstring {

    public static int lengthOfLongestSubstring(String s) {
        if (s.length() == 1) {
            return 1;
        }

        String target = "";

        String mid = "";

        for (int i = 0; i < s.length(); i++) {
            String is = String.valueOf(s.charAt(i));
            if (mid.contains(is)) {

                if (mid.length() > target.length()) {
                    target = mid;
                }
                mid = mid.substring(mid.indexOf(is) + 1) + is;
            } else {
                mid = mid + is;
            }
        }
        return Math.max(mid.length(), target.length());
    }

    public static void main(String[] args) {
        System.out.println(lengthOfLongestSubstring("vddfc"));
    }
}
